### PR TITLE
Cache results of is_torch_tpu_available()

### PR DESCRIPTION
**Pull request by comaniac**
# What does this PR do?

`xm.xla_device()` (called by `is_torch_tpu_available()`) hangs when calling multiple times but no XLA devices are available, and this results in Trainer hanging. Since currently `torch_xla` will be used as long as it is installed in the current active Python environment, I encountered this issue even when I only want to run the Trainer with PyTorch on GPU.

The detail reason behind `torch_xla` is still under investigation (see https://github.com/pytorch/xla/issues/3939).

To workaround this issue, this PR adds `lru_cache` to `is_torch_tpu_available()`, so that `xm.xla_device()` is guaranteed to be called only once when no XLA device is available.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@muellerzr @sgugger

### CODE REVIEW

File: src/transformers/utils/import_utils.py [hunk 1]
```diff
@@ -414,6 +414,7 @@ def is_ftfy_available():
     return _ftfy_available
 
 
+@lru_cache()
```
Comment (comaniac) [inline src/transformers/utils/import_utils.py:417]:
```suggestion
@lru_cache(max_size=2)
```

File: src/transformers/utils/import_utils.py [hunk 2]
```diff
@@ -414,6 +414,7 @@ def is_ftfy_available():
     return _ftfy_available
 
 
+@lru_cache(max_size=2)
```
Comment (comaniac) [inline src/transformers/utils/import_utils.py:417]:
```suggestion
@lru_cache()
```

**Comment by comaniac** [comment]
Not sure the reason of CI failure. It seems not relevant to this PR.

**Review by LysandreJik — APPROVED** [review]
This sounds fair to me, but I'll let @sgugger, mastermind of the Trainer, review when he's back from leave at the end of the week!

**Comment by comaniac** [comment]
@LysandreJik thanks for the review and sure we could wait for @sgugger.
Meanwhile, do I need to do anything to fix the CI failure?

**Review by sgugger — APPROVED** [review]
Thanks a lot for your PR! Tests failures seem unrelated, I have tried re-runing them. Could you you rebase on main if the failure persist?

**Comment by sgugger** [comment]
Thanks for bearing with us. The test failures are spurious and unrelated, so we can merge this.

**Pull request merged**